File: Triatlon-C#/TriatlonPersistance/databases/ProbaDB.cs
using System;
using System.Collections.Generic;
using System.Data.SQLite;
using System.Windows.Forms;
using log4net;
using Triatlon.repository.interfaces;
using TriatlonModel;

namespace ConcursTriatlon.repository.databases
{
    public class ProbaDB : ProbaRepository
    {
        public static readonly ILog log = LogManager.GetLogger("ProbaDB");
        private string connectionString;
        
        public ProbaDB(string connectionString)
        {
            log.Info("Creating ProbaDB");
            this.connectionString = connectionString;
        }
        
        public Proba findOne(long id)
        {
            log.InfoFormat("Entering findOne with value {0}", id);
            using (SQLiteConnection connection = new SQLiteConnection(connectionString))
            {
                try
                {
                    connection.Open();
                    SQLiteCommand command = new SQLiteCommand("select * from proba where id=@id", connection);
                    command.Parameters.AddWithValue("@id", id);
                    SQLiteDataReader reader = command.ExecuteReader();
                    while (reader.Read())
                    {
                        string nume = reader.GetString(1);
                        long idArbitru = reader.GetInt64(2);
                        Proba proba = new Proba(nume,idArbitru);
                        proba.setId(id);
                        reader.Close();

                        return proba;
                    }

                }
                catch (Exception e)
                {
                    log.Error(e.Message);
                    MessageBox.Show(e.Message);
                }
            }
            log.InfoFormat("Exiting findOne with value {0}", null);
            return null;
        }


        public Dictionary<Participant, long> GetParticipantiProba(Proba proba)
        {
            log.InfoFormat("Entering GetParticipantiProba with value {0}", proba);
            Dictionary<Participant, long> participanti = new Dictionary<Participant, long>();
            long idProba = proba.getId();
            using (SQLiteConnection connection = new SQLiteConnection(connectionString))
            {
                try
                {
                    connection.Open();
                    SQLiteCommand command = new SQLiteCommand("select pa.id, pa.nume, pa.prenume, insc.punctaj from participant pa inner join inscriere insc on pa.id = insc.id_participant where insc.id_proba = @id_proba and insc.punctaj > 0 order by insc.punctaj desc", connection);
                    command.Parameters.AddWithValue("@id_proba", idProba);
                    SQLiteDataReader reader = command.ExecuteReader();
                    while (reader.Read())
                    {
                        long idParticipant = reader.GetInt64(0);
                        string nume = reader.GetString(1);
                        string prenume = reader.GetString(2);
                        long punctaj = reader.GetInt64(3);
                        Participant participant = new Participant(nume, prenume);
                        participant.setId(idParticipant);
                        participanti.Add(participant, punctaj);
                    }
                    reader.Close();

                    connection.Close();
                } catch (Exception e)
                {
                    log.Error(e.Message);
                    MessageBox.Show(e.Message);
                }
            }
            log.InfoFormat("Exiting GetParticipantiProba with value {0}", participanti);
            return participanti;
        }

        public IEnumerable<Proba> findAll()
        {
            log.Info("Entering findAll");
            List<Proba> probe = new List<Proba>();
            using (SQLiteConnection connection = new SQLiteConnection(connectionString))
            {
                try
                {
                    connection.Open();
                    SQLiteCommand command = new SQLiteCommand("select * from proba", connection);
                    SQLiteDataReader reader = command.ExecuteReader();
                    while (reader.Read())
                    {
                        long id = reader.GetInt64(0);
                        string nume = reader.GetString(1);
                        long idArbitru = reader.GetInt64(2);
                        Proba proba = new Proba(nume,idArbitru);
                        proba.setId(id);
                        probe.Add(proba);
                    }
                    reader.Close();
                } catch (Exception e)
                {
                    log.Error(e.Message);
                    MessageBox.Show(e.Message);
                }
                connection.Close();
            }
            log.InfoFormat("Exiting findAll with value {0}", probe);
            return probe;
        }

        public Proba save(Proba entity)
        {
            log.InfoFormat("Entering save with value {0}", entity);
            using (SQLiteConnection connection = new SQLiteConnection(connectionString))
            {
                try
                {
                    connection.Open();
                    SQLiteCommand command = new SQLiteCommand("insert into proba(nume, id_arbitru) values (@nume, @idArbitru)", connection);
                    command.Parameters.AddWithValue("@nume", entity.NumeProba);
                    command.Parameters.AddWithValue("@idArbitru", entity.IdArbitru);
                    
                    command.ExecuteNonQuery();
                    
                } catch (Exception e)
                {
                    log.Error(e.Message);
                    MessageBox.Show(e.Message);
                }
            }
            log.InfoFormat("Exiting save with value {0}", entity);
            return entity;
        }

        public Proba delete(long id)
        {
            return null;
        }

        public Proba update(Proba entity1, Proba entity2)
        {
            return null;
        }

        
    }
}

File: Triatlon-Java/TriatlonPersistance/src/main/java/repository/databases/ProbaDB.java
package repository.databases;

import domain.*;
import repository.Utils.JdbcUtils;
import repository.interfaces.ProbaRepository;
import org.apache.logging.log4j.LogManager;
import org.apache.logging.log4j.Logger;

import java.io.FileNotFoundException;
import java.sql.Connection;
import java.sql.PreparedStatement;
import java.sql.ResultSet;
import java.sql.SQLException;
import java.util.*;

public class ProbaDB implements ProbaRepository {
    private final JdbcUtils jdbcUtils;

    private static final Logger logger = LogManager.getLogger();

    public ProbaDB(Properties props) {
        logger.info("Initializing ProbaDB with properties: {} ", props);
        jdbcUtils = new JdbcUtils(props);
    }


    @Override
    public Proba findOne(Long aLong) {
        return null;
    }

    @Override
    public Iterable<Proba> findAll() {
        logger.traceEntry();
        Connection con = jdbcUtils.getConnection();
        List<Proba> probe = new ArrayList<>();
        try(PreparedStatement preStmt = con.prepareStatement("select * from proba")) {
            try(ResultSet result = preStmt.executeQuery()) {
                while(result.next()) {

                    Long id = result.getLong("id");
                    String nume = result.getString("nume");
                    Long id_arbitru = result.getLong("id_arbitru");

                    Proba proba = new Proba(nume, id_arbitru);
                    proba.setId(id);
                    probe.add(proba);
                }
            }
        } catch (SQLException e) {
            logger.error(e);
            System.err.println("Error DB " + e);
        }
        logger.traceExit(probe);
        return probe;
    }

    @Override
    public Proba save(Proba entity) throws FileNotFoundException {
        logger.traceEntry("saving arbitru {} ", entity);
        Connection con = jdbcUtils.getConnection();
        try(PreparedStatement preparedStatement = con.prepareStatement("insert into proba (nume, id_arbitru) values (?,?)")) {
            preparedStatement.setString(1, entity.getNumeProba());
            preparedStatement.setLong(2, entity.getIdArbitru());


            int result = preparedStatement.executeUpdate();
            logger.trace("Saved {} instances", result);
        } catch (Exception e) {
            logger.error(e);
            System.err.println("Error DB " + e);
        }
        logger.traceExit();
        return null;
    }

    @Override
    public Proba delete(Long aLong) {
        return null;
    }

    @Override
    public Proba update(Proba entity1, Proba entity2) {
        return null;
    }

    @Override
    public Map<Participant, Long> getParticipantiProba(Proba proba) {
        logger.traceEntry("finding participanti with {}", proba);
        Connection con = jdbcUtils.getConnection();
        Map<Participant, Long> participanti = new HashMap<>();
        Long idProba = proba.getId();
        System.out.println(idProba);
        try(PreparedStatement preStmt = con.prepareStatement("select pa.id, pa.nume, pa.prenume, insc.punctaj from participant pa inner join inscriere insc on pa.id = insc.id_participant where insc.id_proba = ? and insc.punctaj > 0 order by insc.punctaj desc")) {
            preStmt.setLong(1, idProba);
            try(ResultSet result = preStmt.executeQuery()) {
                while(result.next()) {
                    Long id = result.getLong("id");
                    String nume = result.getString("nume");
                    String prenume = result.getString("prenume");
                    Long punctaj = result.getLong("punctaj");
                    Participant participant = new Participant(nume, prenume);
                    participant.setId(id);
                    System.out.println(participant);
                    participanti.put(participant, punctaj);
                    System.out.println(participant + " " + punctaj);
                }
            } catch (SQLException e) {
                logger.error(e);
                System.err.println("Error DB " + e);
            }
        } catch (SQLException e) {
            logger.error(e);
            System.err.println("Error DB " + e);
        }
        logger.traceExit(participanti);
        return participanti;
    }
}
